File: books_webflux/src/main/java/ru/zakat/books_webflux/model/Book.java
package ru.zakat.books_webflux.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Table("books")
public class Book {

    @Id
    @Column("book_id")
    private Long id;

    private String author;
    private String title;
    private Double rating;
}

File: rsocket-app/src/main/kotlin/com/zakat/rsocket_app/model/Book.kt
package com.zakat.rsocket_app.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("books")
data class Book(
    @Id
    val id: String?,
    val title: String,
    val publishYear: Int,
    val rating: Double,
)
